rfcs/proposed/task_group_dynamic_dependencies/implementation_details.md

```diff
@@ -0,0 +1,562 @@
+
+The next step is to notify all successors in the list of the task's completion. This is done by traversing the list and decrementing the reference counter
+in each stored `continuation_vertex` and destroying the node.
+
+If the reference counter in a `continuation_vertex` reaches zero, the associated task is returned, allowing the
+predecessor to bypass it by returning it from `execute`. 
+
+If multiple successors are ready to be scheduled, the first ready task is bypassed, and the others are spawned:
+
+`function_task::execute` receives the last-added successor task and combines it with the task returned from the body.
+
+If the body does not return a task, or if the returned task has dependencies, the successor task is bypassed. Otherwise, the task returned by the
+body is bypassed.
+
+### Submitting a task for execution
+
+The implementation of submission functions that accept a `task_handle` has also been updated to support tasks with dependencies. The following functions are
+modified:
+
+* `task_group::run(tbb::task_handle&& t)`
+* `task_group::run_and_wait(tbb::task_handle&& t)`
+* `task_arena::enqueue(tbb::task_handle&& t)`
+* `this_task_arena::enqueue(tbb::task_handle&& t)`
+
+Previously, these functions unconditionally spawned the task owned by `task_handle`.
+
+The dynamic dependencies feature allows a `tbb::task_handle` to own a task that cannot be spawned due to dependencies. It must check whether the owned task
+has any dependencies (i.e., whether `m_continuation_vertex` is initialized) and, if so, decrement the reference counter in the vertex.
+
+If the dynamic state is not initialized for the task, it cannot have any dependencies and is spawned immediately.
+
+If the reference counter reaches zero, the task is ready to be spawned. Otherwise, the task will be spawned by that last predecessor that releases its list
+of successors.
+
+## Transferring completion to the other task
+
+The API `tbb::task_group::transfer_this_task_completion_to(tbb::task_handle& new_task)` works by setting `m_new_dynamic_state` in
+the dynamic state of the currently executing task to point to the dynamic state of `new_task`.
+
+To determine the dynamic state associated with the currently executing task, a pointer to the task must be obtained from the task scheduler.
+To achieve this, the new entry point `tbb::task* r1::get_current_task()` is added. 
+
+`transfer_this_task_completion_to` transfers the successors from the currently executing task to `new_task`.
+If the dynamic state of the currently executing task is not initialized, there cannot be any successors associated with the task.
+
+While transferring the successors, the `m_new_dynamic_state` pointer in the dynamic state is set to `new_task`'s dynamic state. The successors
+are retrieved from the successor list and are appended to the `new_dynamic_state`'s successor list:
+
+```cpp
+void transfer_successors_to(task_dynamic_state* new_dynamic_state) {
```

Comment:
It might be useful to replace the remaining code blocks with the flow diagram for transitioning task completion.

Reply:
Done